src/main/java/com/dataloom/data/DataApi.java

```diff
@@ -81,30 +82,42 @@ Void storeEntityData(
 
     /**
      * Creates a new set of entities.
+     * 
      * @param entitySetId The id of the entity set to write to.
      * @param syncId A time-uuid retrieved from data source api.
      * @param entities A map describe the entities to create. Each key will be used as the entity id and must be unique
-     *                 and stable across repeated integrations of data. If either constraint is violated then data may
-     *                 be overwritten or duplicated.
+     *            and stable across repeated integrations of data. If either constraint is violated then data may be
+     *            overwritten or duplicated.
      * @return
      */
     @PUT( BASE + "/" + ENTITY_DATA + "/" + SET_ID_PATH + "/" + SYNC_ID_PATH )
     Void createEntityData(
             @Path( SET_ID ) UUID entitySetId,
             @Path( SYNC_ID ) UUID syncId,
             @Body Map<String, SetMultimap<UUID, Object>> entities );
-    
+
     /**
-     * Creates a new set of entities for the latest syncId.
+     * Creates a new set of entities for the specified entity set's latest sync id.
+     * 
      * @param entitySetId The id of the entity set to write to.
      * @param entities A map describe the entities to create. Each key will be used as the entity id and must be unique
-     *                 and stable across repeated integrations of data. If either constraint is violated then data may
-     *                 be overwritten or duplicated.
+     *            and stable across repeated integrations of data. If either constraint is violated then data may be
+     *            overwritten or duplicated.
      * @return
      */
     @PUT( BASE + "/" + ENTITY_DATA + "/" + SET_ID_PATH )
     Void createEntityData(
             @Path( SET_ID ) UUID entitySetId,
             @Body Map<String, SetMultimap<UUID, Object>> entities );
 
+    /**
+     * Generates a new sync id for an entity set.
+     * 
+     * @param entitySetId The id of the entity set to generate a new sync id for.
+     * @return A new time-uuid generated in data source api.
+     */
+    @GET( BASE + "/" + SET_ID_PATH + "/syncId" )
+    UUID getNewSyncId(
```

Comment:
I would rename to acquireSyncId and a getLatest and releaseSyncId. Consider creating a separate sync controller even.